partiql-logical/src/lib.rs

```diff
@@ -1,14 +1,53 @@
 use partiql_value::{BindingsName, Value};
-use petgraph::prelude::StableGraph;
-use petgraph::Directed;
 use std::collections::HashMap;
 
+#[derive(Eq, PartialEq, Debug, Clone, Copy, Hash)]
+pub struct OpId(usize);
+
+impl OpId {
+    pub fn index(&self) -> usize {
+        self.0
+    }
+}
+
 #[derive(Debug)]
-pub struct LogicalPlan(pub StableGraph<BindingsExpr, (), Directed>);
+pub struct LogicalPlan<T> {
+    nodes: Vec<T>,
+    edges: Vec<(OpId, OpId)>,
+}
 
-impl LogicalPlan {
+impl<T> LogicalPlan<T> {
     pub fn new() -> Self {
-        LogicalPlan(StableGraph::<BindingsExpr, (), Directed>::new())
+        LogicalPlan {
+            nodes: vec![],
+            edges: vec![],
+        }
+    }
+
+    pub fn add_operator(&mut self, op: T) -> OpId {
+        self.nodes.push(op);
+        OpId(self.operator_count())
+    }
+
+    pub fn add_flow(&mut self, src: OpId, dst: OpId) {
+        let src_idx = src.index() - 1;
+        let dst_idx = dst.index() - 1;
+        assert!(src_idx <= self.operator_count());
```

Comment:
I believe
- created `OpId`s are 1-indexed (they are created after the new node is `push`ed). 
- `operator_count()` is also '1-indexed' 

so either the `-1` is unnecessary or we should replace `<=` with `<`

Reply:
Agreed, I overlooked this as I was rushing to push the commit; will change in the PR that I'm putting together for moving the rest of the operators to DAG plan.